.github/workflows/semver-checks.yml

```diff
@@ -0,0 +1,45 @@
+name: semver-checks
+
+on: [push, pull_request]
+
+env:
+  CARGO_TERM_COLOR: always
+  RUST_BACKTRACE: 1
+
+permissions:
+  issues: write
+  pull-requests: write
+
+jobs:
+  check_if_pr_breaks_semver:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        name: checkout full rep
+        with:
+          fetch-depth: 0
+      - name: Install minimal stable
+        uses: actions-rs/toolchain@v1
+        with:
+          profile: default
+          toolchain: stable
+          override: true
+      - uses: Swatinem/rust-cache@v2
+      - name: Install cargo-semver-checks
+        shell: bash
+        run: |
+          cargo install cargo-semver-checks --locked
+      - name: Run check
+        shell: bash
+        run: |
+          cargo semver-checks --all-features --baseline-rev ${{ github.event.pull_request.base.sha }}
+      - name: On Failure
+        if: failure()
```

Comment:
curious: what is `failure()`/`success()`?

Reply:
these are [builtin GitHub Actions](https://docs.github.com/en/actions/writing-workflows/choosing-what-your-workflow-does/evaluate-expressions-in-workflows-and-actions#status-check-functions) which is actually terrifyingly turing complete YAML :laughing:

Reply:
nice thanks haha sorry didn't really do my homework on that one :)